Dockerfile

```diff
@@ -0,0 +1,42 @@
+
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+ARG py_version=3.8
+FROM apache/beam_python${py_version}_sdk:2.40.0 as beam_sdk
+FROM continuumio/miniconda3:4.12.0
+
+# Update miniconda
+RUN conda update conda -y
+
+# Create conda env using environment.yml
+COPY environment.yml /tmp/environment.yml
+RUN conda env create -f /tmp/environment.yml
+
+# Activate the conda env and update the PATH
+ARG CONDA_ENV_NAME=weather-tools
+RUN echo "source activate ${CONDA_ENV_NAME}" >> ~/.bashrc
+ENV PATH /opt/conda/envs/${CONDA_ENV_NAME}/bin:$PATH
+
+# Install SDK.
+RUN pip install --no-cache-dir apache-beam[gcp]==2.40.0
+
+# Verify that the image does not have conflicting dependencies.
+RUN pip check
```

Comment:
similarly, we may want to `conda` check (whatever the equivalent cmd is).

Reply:
Does conda actually have an equivalent of `pip check`? I did a quick search but couldn't find anything. One alternative would be `/path/to/condas/pip check`

Reply:
IMO this wouldn't even be necessary when the deps are resolved by Conda, is that correct?

Reply:
You're totally right. We can just get rid of this line (conda will use its resolver).